lib/IRGen/GenMeta.cpp

```diff
@@ -1812,19 +1821,36 @@ static void emitInitializeFieldOffsetVector(IRGenFunction &IGF,
   if (auto *classDecl = dyn_cast<ClassDecl>(target)) {
     // Compute class layout flags.
     ClassLayoutFlags flags = ClassLayoutFlags::Swift5Algorithm;
-    if (!doesClassMetadataRequireRelocation(IGM, classDecl))
+
+    switch (IGM.getClassMetadataStrategy(classDecl)) {
+    case ClassMetadataStrategy::Resilient:
+      break;
+
+    case ClassMetadataStrategy::Singleton:
+    case ClassMetadataStrategy::Update:
+    case ClassMetadataStrategy::FixedOrUpdate:
       flags |= ClassLayoutFlags::HasStaticVTable;
```

Comment:
This seems like a case where coalescing "generic" with "resilient ancestry" is hurting us. Generic classes still have static vtables, no?

Reply:
When a generic class is instantiated, we allocate and create a new class metadata from whole cloth. It has the right size but the vtable has to be filled in at runtime.

If you recall a while ago generics were implemented by IRGen emitting a “pattern”; the runtime would memcpy the pattern to a new allocation and patch up a few things. However we don’t do it that way; there is still a pattern but it only has a few things in it and it’s true const.